src/state/sagas/datasets/QueryDatasetTwingraph/QueryDatasetTwingraph.js

```diff
@@ -44,8 +44,16 @@ function* runDatasetTwingraphQuery(action, query, attemptsNumber = 0) {
 function* startAllDatasetTwingraphQueries(action) {
   const { workspace } = action.payload;
   const queries = workspace?.webApp?.options?.datasetManager?.queries ?? [];
+  const graph = workspace?.webApp?.options?.datasetManager?.graphIndicators;
+  const categories = workspace?.webApp?.options?.datasetManager?.categories;
+
   yield all(
     queries.map((query, index) => {
+      if (
+        !categories.some((cat) => cat?.kpis.some((kpi) => kpi?.queryId === query.id)) &&
+        !graph.some((g) => g?.queryId === query.id)
+      )
+        return null;
```

Comment:
Is it okay to have `null` items inside the array of effects provided to `yield all`?
Not sure how it will behave, but the doc mentions:
> When we yield an array of effects, the generator is blocked until all the effects are resolved **or as soon as one is rejected**

We should probably double check that a `null` item won't cancel the other queries :slightly_smiling_face:

Reply:
I have tested queries in different orders in the workspace config so as if a null item would cancel the other queries, errors would instantly be visible on the overview kpis.
It's not the case.